airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -134,22 +137,47 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
         return;
       }
 
-      // if the buffer flushes, update the states appropriately.
-      // TODO: ryankfu (if true, this is where bundling up logic to also commit to airbyte_raw table)
-      if (bufferingStrategy.addRecord(stream, message)) {
-        markStatesAsFlushedToTmpDestination();
+      final Optional<BufferFlushType> flushType = bufferingStrategy.addRecord(stream, message);
```

Comment:
Just stating that I still prefer separating the logic of whether or not to `markStateAsFlushedToDestination` to a helper method and using the Optional oneliner rather than the nested `if` statements